### PR TITLE
Bluetooth: controller: Add Direct Test Mode support

### PR DESCRIPTION
**Pull request by cvinayak**

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ll_test.c [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	}
+
+	/* start coarse timer */
+	cntr_start();
+
+	/* Reset Radio h/w */
+	radio_reset();
+	radio_isr_set(isr);
+
+	/* Store value needed in Rx ISR */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_test.c]: Tx/Rx

File: subsys/bluetooth/controller/ll_sw/ll_test.c [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	} else {
+		test_phy = BIT(2);
+		phy_flags = 0;
+	}
+
+	/* Setup resources required by Radio */
+	hf_clock = radio_hf_clock_get();
+	clock_control_on(hf_clock, (void *)1); /* start clock, blocking. */
+
+	/* Setup Radio in Rx */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_test.c]: Tx/Rx

**Review by carlescufi — CHANGES_REQUESTED** [review]
DTM test issues with this patch

**Comment by cvinayak** [comment]
@carlescufi Full PHY-RF tests for all supported PHYs pass now.

File: subsys/bluetooth/controller/hal/nrf5/radio.c [hunk 3]
```diff
@@ -68,16 +68,28 @@ void radio_phy_set(u8_t phy, u8_t flags)
 	case BIT(0):
 	default:
 		mode = RADIO_MODE_MODE_Ble_1Mbit;
+
+#if defined(CONFIG_SOC_NRF52840)
+		*(volatile u32_t *)0x4000173c &= ~0x80000000;
```
Comment (carlescufi) [inline subsys/bluetooth/controller/hal/nrf5/radio.c]: I understand this is a workaround, but can you confirm that the source code where you took it from does not include a comment about its purpose? Same applies to all the similar expressions below

File: subsys/bluetooth/controller/ll_sw/ll_test.c [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+#define CNTR_MIN_DELTA 3
+
+static const u32_t test_sync_word = 0x71764129;
+static u8_t        test_phy;
+static u8_t        test_phy_flags;
+static u16_t       test_num_rx;
+static bool        started;
+
+/* NOTE: The PRBS9 sequence used as packet payload.
+ * The bytes in the sequence is in the right order, but the bits of each byte
```
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/ll_test.c]: s/is/are

File: subsys/bluetooth/controller/ll_sw/ll_test.c [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+ * in the array is reverse of that found by running the PRBS9 algorithm. This
```
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/ll_test.c]: /is reverse of/are reversed from

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**